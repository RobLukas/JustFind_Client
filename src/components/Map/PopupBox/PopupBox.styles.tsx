import styled from 'styled-components';

const PopupBox = styled.div`
  display: flex;
  padding: 10px 20px;
  cursor: pointer;
  font-size: 13px;
  margin-top: 0;
  margin: 0;
`;

const LogoWrapper = styled.div`
  position: relative;
  width: 40px;
  display: flex;
  flex-shrink: 0;
`;

const CompanyLogo = styled.img`
  margin: auto;
  max-width: 40px;
  max-height: 40px;
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
`;

const DescWrapper = styled.div`
  flex-direction: column;
  display: flex;
  min-width: 140px;
  margin-left: 7px;
  text-align: left;
`;

const CompanyName = styled.span`
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
`;

const JobTitle = CompanyName;

const Salary = styled.span`
  color: #1ec66c;
  font-weight: 400;
  white-space: nowrap;
`;

export default {
  PopupBox,
  CompanyName,
  LogoWrapper,
  JobTitle,
  CompanyLogo,
  DescWrapper,
  Salary,
};
