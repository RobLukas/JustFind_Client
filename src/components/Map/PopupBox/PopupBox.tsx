import React, { FC } from 'react';
import S from './PopupBox.styles';
import PopupProps from 'api/offers.interface';

const PopupBox: FC<PopupProps> = ({
  logoSrc,
  title,
  salary,
  companyName,
}: PopupProps) => {
  return (
    <S.PopupBox>
      <S.LogoWrapper>
        <S.CompanyLogo src={logoSrc} alt={companyName} />
      </S.LogoWrapper>
      <S.DescWrapper>
        <S.JobTitle>{title}</S.JobTitle>
        <S.Salary>{`${salary.from} - ${salary.to} ${salary.currency}`}</S.Salary>
        <S.CompanyName>{companyName}</S.CompanyName>
      </S.DescWrapper>
    </S.PopupBox>
  );
};

export default PopupBox;
