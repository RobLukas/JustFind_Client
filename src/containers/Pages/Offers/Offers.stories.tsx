import React from 'react';
import { Meta, Story } from '@storybook/react/types-6-0';
import Offers from './Offers';

export default {
  title: 'pages/Offers',
  component: Offers,
} as Meta;

const Template: Story = () => <Offers />;

export const Default = Template.bind({});
