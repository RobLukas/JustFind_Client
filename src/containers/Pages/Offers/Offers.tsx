import React, { FC } from 'react';

import { Header, Filters, Map, OffersList } from 'containers';
import offers from 'api/offers';

const Offers: FC = () => {
  return (
    <>
      <Header />
      <Filters />
      <OffersList offers={offers} />
      <Map />
    </>
  );
};

export default Offers;
