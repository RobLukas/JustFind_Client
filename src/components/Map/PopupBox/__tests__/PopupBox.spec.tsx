import React from 'react';
import { cleanup, render } from '@testing-library/react';
import 'jest-styled-components';

import PopupBox from '../PopupBox';
import Offers from 'api/offers.interface';

describe('<PopupBox />', () => {
  afterEach(cleanup);
  it('render correctly', () => {
    const props: Offers = {
      mainlyTechnology: 'JS',
      logoSrc: 'logosrc',
      title: 'title',
      salary: {
        from: 1000,
        to: 2000,
        currency: 'PLN',
      },
      companyName: 'company name',
      companyAddress: 'company address',
      technologies: ['html', 'react', 'css'],
      geoPosition: [1000, 1000],
      createdAt: new Date(),
    };
    const { container } = render(<PopupBox {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
