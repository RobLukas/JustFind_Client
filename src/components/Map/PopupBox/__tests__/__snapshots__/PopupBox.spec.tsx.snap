// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PopupBox /> render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 10px 20px;
  cursor: pointer;
  font-size: 13px;
  margin-top: 0;
  margin: 0;
}

.c1 {
  position: relative;
  width: 40px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c2 {
  margin: auto;
  max-width: 40px;
  max-height: 40px;
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
}

.c3 {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-width: 140px;
  margin-left: 7px;
  text-align: left;
}

.c4 {
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
}

.c5 {
  color: #1ec66c;
  font-weight: 400;
  white-space: nowrap;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <img
      alt="company name"
      class="c2"
      src="logosrc"
    />
  </div>
  <div
    class="c3"
  >
    <span
      class="c4"
    >
      title
    </span>
    <span
      class="c5"
    >
      1000 - 2000 PLN
    </span>
    <span
      class="c4"
    >
      company name
    </span>
  </div>
</div>
`;
